src/OpenLoco/src/Ui/Windows/Construction/Construction.h

```diff
@@ -172,18 +173,22 @@ namespace OpenLoco::Ui::Windows::Construction
         void setTrackOptions(const uint8_t trackType);
         void setDisabledWidgets(Window* self);
         void createConstructionWindow();
-        void refreshAirportList(uint8_t* stationList);
-        void refreshDockList(uint8_t* stationList);
-        void refreshStationList(uint8_t* stationList, uint8_t trackType, TransportMode transportMode);
-        void refreshBridgeList(uint8_t* bridgeList, uint8_t trackType, TransportMode transportMode);
-        void refreshModList(uint8_t* modList, uint8_t trackType, TransportMode transportMode);
         void sub_4A3A50();
-        void refreshSignalList(uint8_t* signalList, uint8_t trackType);
         void setNextAndPreviousTrackTile(const TrackElement& elTrack, const World::Pos2& pos);
         void setNextAndPreviousRoadTile(const RoadElement& elRoad, const World::Pos2& pos);
         bool isPointCloserToNextOrPreviousTile(const Point& point, const Viewport& viewport);
         void previousTab(Window* self);
         void nextTab(Window* self);
+
+        template<typename T>
+        void copyToLegacyList(T sflType, uint8_t* legacyList)
```

Comment:
```suggestion
        void copyToLegacyList(const T& sflType, uint8_t* legacyList)
```
Also should we perhaps pass a max size argument here to ensure it never goes out of bounds?

Reply:
This is hopefully just a temporary function that will go away soon